src/libostree/ostree-repo-static-delta-processing.c

```diff
@@ -423,14 +456,48 @@ dispatch_bspatch (OstreeRepo *repo, StaticDeltaExecutionState *state, GCancellab
       struct bspatch_stream stream;
       stream.read = bspatch_read;
       stream.opaque = &opaque;
-      if (bspatch ((const guint8 *)g_mapped_file_get_contents (input_mfile),
-                   g_mapped_file_get_length (input_mfile), buf, state->content_size, &stream)
-          < 0)
+
+      int bspatch_result
+          = bspatch ((const guint8 *)g_mapped_file_get_contents (input_mfile),
+                     g_mapped_file_get_length (input_mfile), buf, state->content_size, &stream);
+
+      /* Unmap before checking result - only if we actually created a mapping */
+      if (state->content_size > 0 && munmap (buf, state->content_size) < 0)
+        return glnx_throw_errno_prefix (error, "munmap");
+
+      if (bspatch_result < 0)
         return glnx_throw (error, "bsdiff patch failed");
 
-      if (!_ostree_repo_bare_content_write (repo, &state->content_out, buf, state->content_size,
-                                            cancellable, error))
-        return FALSE;
+      /* Now read from tmpfile and write to repository (if content_size > 0) */
```

Comment:
Why was this done? Previously, the (potentially) large output buffer was used to be written into the repository via `_ostree_repo_bare_content_write`. Now, this code loads the patched content chunk-based into memory and then writes it to disk. But, as the source was changed to be potentially memory-backed, this is not needed - I would even say it introduced an unnecessary copy?

Furthermore, the `glnx_loop_write` part of `_ostree_repo_bare_content_write` now may performs additional calls to `write`, as the chuck sizes may not align with its retry-logic as from https://github.com/GNOME/libglnx/blob/de29c5f7d9df8d57b4f5caa9920f5d4caa7a8cfc/glnx-fdio.c#L752